frontend/src/pages/SignUp.jsx

```diff
@@ -1,11 +1,77 @@
-const SignUp = () => {
-    return(
-      <div>
-        <p>
-          Sign Up Page
-        </p>
-      </div>
-    )
-  }
-  
-  export default SignUp
\ No newline at end of file
+import React from "react";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from "../contexts/AuthContext";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default function SignUp() {
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [confirmPassword, setConfirmPassword] = useState("");
+	const [loading, setLoading] = useState(false);
+	const navigate = useNavigate();
+	const { session, register } = useAuth();
+
+	useEffect(() => {
+		if (session) {
+			navigate("/dashboard");
+		}
+	}, [session]);
+
+	async function handleSubmit(e) {
+		e.preventDefault();
+		if (password !== confirmPassword) {
+			return alert("Passwords do not match");
+		}
+
```

Comment:
Need more error handling cases here (in case email is not valid, email is already registered with an account, etc), and the alert should display the particular error, instead of just "Failed to Register"

Reply:
- Added general from supabase if doesn't register correctly -- if user is already registered, if passwords do not fit the criteria, etc...